larky/src/main/resources/vendor/OpenPGP/Crypto.star

```diff
@@ -0,0 +1,533 @@
+load("@stdlib//struct", struct="struct")
+load("@stdlib//types", types="types")
+load("@vendor//Crypto/Random", Random="Random")
+load("@stdlib//builtins", builtins="builtins")
+# load("@vendor//Crypto/Cipher", DES3="DES3")
+# load("@vendor//Crypto/Cipher", Blowfish="Blowfish")
+load("@vendor//Crypto/Cipher/PKCS1_v1_5", PKCS1_v1_5_Cipher="PKCS1_v1_5_Cipher")
+# load("@vendor//Crypto/Cipher", CAST="CAST")
+load("@vendor//Crypto/Cipher/AES", AES="AES")
+# load("@vendor//Crypto/Hash", SHA512="SHA512")
+# load("@vendor//Crypto/Hash", MD5="MD5")
+# load("@vendor//Crypto/Hash", SHA224="SHA224")
+# load("@vendor//Crypto/Hash", SHA384="SHA384")
+# load("@vendor//Crypto/Hash", RIPEMD="RIPEMD")
+load("@vendor//Crypto/Hash/SHA", SHA="SHA")
+# load("@vendor//Crypto/Hash", SHA256="SHA256")
+# load("@vendor//Crypto/PublicKey", RSA="RSA")
+# load("@vendor//Crypto/PublicKey", DSA="DSA")
+# load("@vendor//Crypto/Random", random="random")
+# load("@vendor//Crypto/Signature", PKCS1_v1_5="PKCS1_v1_5")
+# load("@vendor//Crypto/Util", number="number")
+load("@vendor//OpenPGP", OpenPGP="OpenPGP")
+load("@vendor//option/result", Error="Error")
+
+pack = struct.pack
+unpack = struct.unpack
+
+__all__ = ['Wrapper']
+def Wrapper(packet):
+    """ A wrapper for using the classes from OpenPGP.py with PyCrypto """
+    self = larky.mutablestruct(__name__='Wrapper', __class__=Wrapper)
+
+    # def _parse_packet(packet):
+    #     if types.is_instance(packet, OpenPGP.Packet) or types.is_instance(packet, OpenPGP.Message) or types.is_instance(packet, Crypto.PublicKey.RSA._RSAobj) or types.is_instance(packet, Crypto.PublicKey.DSA._DSAobj):
+    #         return packet
+    #     elif types.is_tuple(packet) or types.is_list(packet):
+    #         if sys.version_info[0] == 2 and types.is_instance(packet[0], long) or types.is_int(packet[0]):
+    #             data = []
+    #             for i in packet:
+    #                 data.append(long_to_bytes(i)) # OpenPGP likes bytes
+    #         else:
+    #             data = packet
+    #         data = packet
+    #         return OpenPGP.SecretKeyPacket(keydata=data, algorithm=1, version=3) # V3 for fingerprint with no timestamp
+    #     else:
+    #         return OpenPGP.Message.parse(packet)
+    # self._parse_packet = _parse_packet
+
+    def __init__(packet):
+        # packet = self._parse_packet(packet)
+        self._key = None
+        self._message = self._key
+        if types.is_instance(packet, OpenPGP.PublicKeyPacket) or (hasattr(packet, '__getitem__') and types.is_instance(packet[0], OpenPGP.PublicKeyPacket)): 
+        # If it's a key (other keys are subclasses of this one)
+            self._key = packet
+        else:
+            self._message = packet
+        return self
+    self = __init__(packet)
+
+    def get_cipher(algo):
+        def cipher(m, ks, bs):
+                def _cipher(k, iv):
+                    return m.new(k, mode=AES.MODE_CFB,
+                        IV=iv or b'\0'*bs,
```

Comment:
Using a predictable or repeated IV with AES CFB is not secure.  It looks to me like we need to enforce a unique IV here

Reply:
I am semi wrong here, I was originally thinking of CBC.  I just checked and this is is not recommended by NIST, but may be secure.  Still something to carefully consider